server/app/controllers/applicant/ApplicantProgramsController.java

```diff
@@ -68,56 +61,16 @@ public CompletionStage<Result> index(Request request, long applicantId) {
     return applicantStage
         .thenComposeAsync(v -> checkApplicantAuthorization(profileUtils, request, applicantId))
         .thenComposeAsync(
-            v -> applicantService.relevantPrograms(applicantId), httpContext.current())
+            v -> applicantService.relevantProgramsForApplicant(applicantId), httpContext.current())
         .thenApplyAsync(
-            allPrograms -> {
-              ImmutableList<ProgramDefinition> programsWithDraftApplications =
-                  allPrograms.get(LifecycleStage.DRAFT);
-              Set<String> adminNames =
-                  programsWithDraftApplications.stream()
-                      .map(ProgramDefinition::adminName)
-                      .collect(Collectors.toSet());
-              ImmutableList<ProgramDefinition> dedupedActivePrograms =
-                  allPrograms.get(LifecycleStage.ACTIVE).stream()
-                      .filter(
-                          programDefinition -> !adminNames.contains(programDefinition.adminName()))
-                      .collect(ImmutableList.toImmutableList());
-              // Deduplicate programs with draft applications. This is a temporary fix for a bug
```

Comment:
Note to the reviewer: This deduplication / logging moved into ApplicantServiceImpl.relevantProgramsForApplicant.